include/array.h

```diff
@@ -8,6 +8,7 @@ namespace quda
    */
   template <typename T, int n> struct array {
     using value_type = T;
+    static constexpr int N = n;
```

Comment:
What purpose does this serve that `array::size()` doesn't

Reply:
It's static, so instantiation isn't necessary.